game/game.go

```diff
@@ -548,8 +582,39 @@ func (g *Game) Leavable() bool {
 	return !g.Started
 }
 
-func (g *Game) Joinable() bool {
-	return !g.Closed && g.NMembers < len(variants.Variants[g.Variant].Nations) && len(g.ActiveBans) == 0 && len(g.FailedRequirements) == 0
+func (g *Game) IsGameMasterInvited(email string) bool {
+	for _, invitation := range g.GameMasterInvitations {
+		if invitation.Email == email {
+			return true
+		}
+	}
+	return false
+}
+
+func (g *Game) HasReplaceableMember() bool {
+	for _, member := range g.Members {
+		if member.Replaceable {
+			return true
+		}
+	}
+	return false
+}
+
+func (g *Game) Joinable(user *auth.User) bool {
+	if len(g.ActiveBans) > 0 || len(g.FailedRequirements) > 0 {
+		return false
+	}
+	if g.Closed || g.NMembers >= len(variants.Variants[g.Variant].Nations) {
+		if g.GameMasterEnabled && g.HasReplaceableMember() && (!g.RequireGameMasterInvitation || g.IsGameMasterInvited(user.Email)) {
+			return true
+		}
+		return false
+	} else {
+		if g.GameMasterEnabled && g.RequireGameMasterInvitation && !g.IsGameMasterInvited(user.Email) {
+			return false
+		}
```

Comment:
This if statement seems redundant - we return false regardless.

Reply:
Ah, the other option should have been true. Good catch :D